sub-packages/bionemo-size-aware-batching/tests/bionemo/size_aware_batching/test_sampler.py

```diff
@@ -494,37 +496,37 @@ def test_iter_bucket_batch_sampler_with_shuffle(sample_data):
         base_batch_sampler_individual_kwargs,
     ) = sample_data
     batch_sampler = BucketBatchSampler(
-        sizes=sizes,
-        bucket_boundaries=bucket_boundaries,
+        sizes=sizes.cuda(),
+        bucket_boundaries=bucket_boundaries.cuda(),
         base_batch_sampler_class=base_batch_sampler_class,
         base_batch_sampler_shared_kwargs=base_batch_sampler_shared_kwargs,
         base_batch_sampler_individual_kwargs=base_batch_sampler_individual_kwargs,
         shuffle=True,
-        generator=torch.Generator().manual_seed(0),
+        generator=torch.Generator(device="cuda").manual_seed(0),
     )
     batch_lists_first_iter = list(iter(batch_sampler))
     ref_batch_lists_first_iter = [
-        [24, 17, 16, 22, 19],
-        [2, 5],
-        [12, 10, 11],
-        [3, 0],
-        [15, 18, 20, 21, 23],
-        [7, 13, 6],
-        [14, 9, 8],
-        [1, 4],
+        [23, 24, 16, 21, 15],
+        [17, 20, 18, 22, 19],
+        [9, 13, 7],
+        [11, 6, 10],
+        [4, 5],
+        [2, 3],
+        [12, 8, 14],
+        [0, 1],
     ]
```

Comment:
why is the data changing?

Reply:
previously, changing torch random call from cpu to gpu will lead to different random state, thus the reference data change.

We updated to avoid numerical check, this part is removed